node/v2/handler.js

```diff
@@ -573,9 +573,17 @@ function sendCallRequestFrame(req, flags, args) {
         self.errorEvent.emit(self, errors.SendCallReqBeforeIdentifiedError());
         return;
     }
+
+    var checksum;
+    if (req.checksum.frozen) {
+        checksum = req.checksum;
+    } else {
+        checksum = req.checksum.type;
```

Comment:
All the checksumming code goes through `ChecksumOrType : (CheckSum | Number) => CheckSum`

So it should be an object or a numeric type.